### ISSUE TITLE
InputEvent needs easier to comprehed writing structure and a lot more practical examples

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot-docs#2311]
https://docs.godotengine.org/en/3.1/tutorials/inputs/inputevent.html

So far I used Godots input maps which are straight forward and easy to use. However now I would like to store all kinds of inputs in arrays and differentiate between button hold and just pressed, and create a list and compare it to other lists for a combo system. For that input maps seem less ideal I so I was looking into input events.

I don't know how many times I've went over and through this page and could distill nothing of use.
Worst of all is the category "How does it work". Those points are written in a language I speak and understand fluidly, but I keep on asking myself: "Ok, now what does that mean in a practical sense?!"
These things need a clearer language and a lot more practical examples. Even more so the more complex a topic is.

**Comment by clayjohn** [comment]
You are right, this document is definitely an advanced description of how InputEvents work. In its current form it is not aimed at teaching new users the basics. It is a technical description for more advanced users. 

It would be very beneficial to have a gentler walkthrough style tutorial that explains how to do some more advanced interaction (including button holds, combos etc.)
